app/services/IdentityService.scala

```diff
@@ -29,28 +27,38 @@ class IdentityService(identityApiClient: IdentityApiClient) extends LazyLogging
       (response.json \ "user").asOpt[IdUser]
     }
 
-  def registerGuest(personalData: PersonalData): Future[Option[UserId]] = {
-
+  def registerGuest(personalData: PersonalData): Future[GuestUser] = {
     val json = JsObject(Map(
       "primaryEmailAddress" -> JsString(personalData.email),
       "privateFields" -> JsObject(Map(
-        "firstName" -> JsString(personalData.firstName),
+        "firstName" ->  JsString(personalData.firstName),
         "secondName" -> JsString(personalData.lastName),
         "billingAddress1" -> JsString(personalData.address.house),
         "billingAddress2" -> JsString(personalData.address.street),
         "billingAddress3" -> JsString(personalData.address.town),
         "billingPostcode" -> JsString(personalData.address.postcode),
         "billingCountry" -> JsString("United Kingdom")
-      ).toSeq),
-      "statusFields" -> JsObject(Map("receiveGnmMarketing" -> JsString("true")).toSeq)
-    ).toSeq)
+      )),
+      "statusFields" -> JsObject(Map("receiveGnmMarketing" -> JsString("true")))
+    ))
+
+    identityApiClient.createGuest(json).map(response => response.json.as[GuestUser])
+  }
 
+  def convertGuest(gu: GuestUser, password: String): Future[Unit] = {
+    val json = JsObject(Map("password" -> JsString(password)))
 
-    identityApiClient.createGuest(json).map(response => (response.json \ "guestRegistrationRequest" \ "userId").asOpt[String].map(UserId))
+    identityApiClient.convertGuest(json, gu).map { response =>
+      if (!response.status.toString.matches("""2\d\d""")) {
```

Comment:
We could match on 201, but I think it would be a bit more robust if we match on something like `2xx` (i.e. any success status code)